02.calendar/calendar.rb

```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env ruby
+require 'date'
+require 'optparse'
+
+params = ARGV.getopts("m:", "y:", "m:#{Date.today.mon}", "y:#{Date.today.year}")
+month = params["m"].to_i
+year = params["y"].to_i
+
+start_date = Date.new(year, month, 1)
+end_date = Date.new(year, month, -1)
+
+header = "#{start_date.mon}月 #{start_date.year}"
+puts header.center(20)
+
+puts "日 月 火 水 木 金 土"
+
+calendar_formatted = (start_date..end_date).map do |date|
+  day_to_string = date.day.to_s.rjust(2)
+  date.day != 1 && date.sunday? ? "\n" + day_to_string : day_to_string
+end
+
+margin_space = Array.new(start_date.wday, "  ")
+
+calendar = margin_space + calendar_formatted
```

Comment:
「マージン + フォーマットされたカレンダー = カレンダー」という構図に違和感があります。
最終的にできあがるものが`calendar`なら、`calendar_formatted`は「まだカレンダーにはなってない何か」になると思います。
つまり`calendar_formatted`は命名に問題があると言えそうです。

たとえばこんな風にするのはどうでしょう？
（マージン + 整形された日々 = カレンダー、みたいなイメージ）

```suggestion
calendar = margin_space + formatted_days
```

Reply:
なるほどです。変数名一つとっても論理的に自然でなければならないのですね。。。
こちら修正しました!!
37d1917b5024e89b02a4f76c933210dfd870bd33